src/main/java/edu/wpi/grip/ui/preview/ImageSocketPreviewView.java

```diff
@@ -0,0 +1,85 @@
+package edu.wpi.grip.ui.preview;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedBytes;
+import edu.wpi.grip.core.Socket;
+import javafx.application.Platform;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.image.PixelFormat;
+import javafx.scene.image.WritableImage;
+
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+
+import static org.bytedeco.javacpp.opencv_core.Mat;
+
+/**
+ * A <code>SocketPreviewView</code> that previews sockets containing OpenCV Mats
+ */
+public class ImageSocketPreviewView extends SocketPreviewView<Mat> {
+
+    private final ImageView imageView;
+    private WritableImage image;
+    private IntBuffer pixels;
+
+    /**
+     * @param eventBus The EventBus used by the application
+     * @param socket   An output socket to preview
+     */
+    public ImageSocketPreviewView(EventBus eventBus, Socket<Mat> socket) {
+        super(eventBus, socket);
+
+        this.convertImage();
+
+
+        this.imageView = new ImageView(this.image);
+        this.setContent(imageView);
+
+        // Every time the Mat changes, convert it into a JavaFX image and show it.
+        this.valueProperty().addListener(observable -> Platform.runLater(() -> {
+            this.convertImage();
+            this.imageView.setImage(this.image);
+        }));
+    }
+
+    /**
+     * Convert a BGR-formatted OpenCV {@link Mat} into a JavaFX {@link Image}.
+     * <p>
+     * JavaFX understands ARGB pixel data, so one way to turn a Mat into a JavaFX image is to shift around the bytes
+     * from the Mat into an int array of pixels.
+     * <p>
+     * This is also possible to do by using JavaCV, but the JavaCV method involves several intermediate conversions
+     * (Mat -> Frame -> BufferedImage -> JavaFX Image) and is way too slow to use for a real-time video.
+     */
+    private void convertImage() {
```

Comment:
It has to be called in the application thread anyways (that's way `Platform.runLater` does)